### ISSUE TITLE
Remove BUFR query code from this project.

### ISSUE THREAD

**Issue by rmclaren** [JCSDA-internal/ioda-converters#1586]
## Description

Need to remove bufr-query code from this project as it has moved to https://github.com/NOAA-EMC/bufr-query. The version of the BUFR querying code in this project is **depricated**. Need to delete the code and all the associated supporting files (test input files, test data files.... etc.).

Basically:

1. Delete src/bufr directory and all its contents. Also test/bufr and tools/bufr etc...
2. Delete all files that match the pattern test/testinput/bufr_*.yaml. Remove those tests from test/CMakeLists.txt.
3. Delete the associated BUFR data files from the test data project or test/testinput directory that are being used by the previous tests.
4. Delete relevant output *.nc files from the test data project and test/testoutput (those files associated with test/CMakeLists.txt).

An external dependency needs to be added so that any scripts that depend on BUFR in this project are able to use that functionality. Those same scripts will need modifications to work with the updated version of the framework... 

Basically:

1. Add external dependency to bufr-query in the main CMakeLists.txt file
2. Modify scripts and yaml files as necessary.

## Requirements

BUFR-query code needs to be removed.

**Comment by BenjaminRuston** [comment]
@emilyhcliu I guess we should remove some of these; however, people (including those at NOAA) do use these

and a number work just fine with the `bufr2ioda.x` ,, so all for removing some deprecated ctests, but simple decodes should still be supported with the yaml calling style I presume

**Comment by rmclaren** [comment]
@BenjaminRuston The code in the bufr-query project works the same with regard to the YAML files (though they may need some minor modifications).

**Comment by BenjaminRuston** [comment]
@rmclaren and @srherbener this may need a revisit, the bigger picture is now the bufr-query is part of the spack-stack.  There are unit test in ioda converters that rely on this do they belong here?  It was a nice capability and would be good to have instructions and a few examples within ioda-converters imho if the plan is to keep it as part of spack stack.  We could pare these down though.
